test/unit/specs/api.spec.js

```diff
@@ -151,38 +151,87 @@ describe('router.push/replace callbacks', () => {
       }, 1)
     })
   })
+  describe('callbacks', () => {
+    it('push complete', done => {
+      router.push('/foo', () => {
+        expect(calls).toEqual([1, 2, 3, 4])
+        done()
+      })
+    })
 
-  it('push complete', done => {
-    router.push('/foo', () => {
-      expect(calls).toEqual([1, 2, 3, 4])
-      done()
+    it('push abort', done => {
+      router.push('/foo', spy1, spy2)
+      router.push('/bar', () => {
+        expect(calls).toEqual([1, 1, 2, 2])
+        expect(spy1).not.toHaveBeenCalled()
+        expect(spy2).toHaveBeenCalled()
+        done()
+      })
     })
-  })
 
-  it('push abort', done => {
-    router.push('/foo', spy1, spy2)
-    router.push('/bar', () => {
-      expect(calls).toEqual([1, 1, 2, 2])
-      expect(spy1).not.toHaveBeenCalled()
-      expect(spy2).toHaveBeenCalled()
-      done()
+    it('replace complete', done => {
+      router.replace('/foo', () => {
+        expect(calls).toEqual([1, 2, 3, 4])
+        done()
+      })
     })
-  })
 
-  it('replace complete', done => {
-    router.replace('/foo', () => {
-      expect(calls).toEqual([1, 2, 3, 4])
-      done()
+    it('replace abort', done => {
+      router.replace('/foo', spy1, spy2)
+      router.replace('/bar', () => {
+        expect(calls).toEqual([1, 1, 2, 2])
+        expect(spy1).not.toHaveBeenCalled()
+        expect(spy2).toHaveBeenCalled()
+        done()
+      })
     })
   })
 
-  it('replace abort', done => {
-    router.replace('/foo', spy1, spy2)
-    router.replace('/bar', () => {
-      expect(calls).toEqual([1, 1, 2, 2])
-      expect(spy1).not.toHaveBeenCalled()
-      expect(spy2).toHaveBeenCalled()
-      done()
+  describe('promises', () => {
+    it('push with callback return undefined', done => {
```

Comment:
This test should be moved to the group of callbacks

```suggestion
    it('does not return a Promise when a callback is passed', done => {
```

Reply:
Done